From fc2202f2cc4b9cc27cf8e25855a8703fc6b998e6 Mon Sep 17 00:00:00 2001
From: Vincent Bernat <vincent@bernat.im>
Date: Sun, 30 Jul 2017 17:27:22 +0200
Subject: [PATCH] netlink: ignore unknown attributes

---
 net/core/rtnetlink.c | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/net/core/rtnetlink.c b/net/core/rtnetlink.c
index fad649ae4dec..91e0c35b411d 100644
--- a/net/core/rtnetlink.c
+++ b/net/core/rtnetlink.c
@@ -2325,9 +2325,8 @@ static int rtnetlink_rcv_msg(struct sk_buff *skb, struct nlmsghdr *nlh)
 		while (RTA_OK(attr, attrlen)) {
 			unsigned int flavor = attr->rta_type;
 			if (flavor) {
-				if (flavor > rta_max[sz_idx])
-					return -EINVAL;
-				rta_buf[flavor-1] = attr;
+				if (flavor <= rta_max[sz_idx])
+					rta_buf[flavor-1] = attr;
 			}
 			attr = RTA_NEXT(attr, attrlen);
 		}
-- 
2.13.3

